### PR TITLE
CC: Enable guest-pull tests on non-TEE for s390x

### PR DESCRIPTION
**Pull request by BbolroC**
This commit is to add a new CI job to run-k8s-tests-on-zvsi.yaml.
Why the job is not configured in run-kata-coco-tests.yaml by having it integrated with `run-k8s-tests-coco-nontee` is:

- It uses k3s instead of AKS
- It runs on a self-hosted runner

These differences make the integrated job not easy to read and maintain when it comes to incorporating other platforms in the near future.

The verification for non-TEE has been done via a private self-hosted runner at https://github.com/BbolroC/kata-containers/actions/runs/8782763328

This PR also includes an adjustment of the build procedure for `boot-image-se` and updates the document accordingly. The verification for TEE (Secure Execution) from the internal test is at https://github.com/kata-containers/kata-containers/actions/runs/8795055047/job/24144779614

Fixes: #9467

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

**IMPORTANT NOTE**: This PR should not be merged before #9489.

### CODE REVIEW

File: .github/workflows/run-k8s-tests-on-zvsi.yaml [hunk 1]
```diff
@@ -78,3 +78,62 @@ jobs:
...
+          - nydus
+        pull-type:
+          - guest-pull
+    runs-on: s390x-large
+    env:
+      DOCKER_REGISTRY: ${{ inputs.registry }}
+      DOCKER_REPO: ${{ inputs.repo }}
+      DOCKER_TAG: ${{ inputs.tag }}
+      GH_PR_NUMBER: ${{ inputs.pr-number }}
+      KATA_HOST_OS: ${{ matrix.host_os }}
```
Comment (stevenhorsman) [inline .github/workflows/run-k8s-tests-on-zvsi.yaml:98]: There doesn't seem to be a `host_os` matrix parameter, so should this `env` be set to `ubuntu` (I guess?) directly?

**Review by stevenhorsman — COMMENTED** [review]
The workflow for `run-k8s-tests-coco-nontee` and `run-k8s-tests` looks pretty much identical apart from the snapshotter and pull type, so I wonder if you could add the nydus snapshotter to the original snapshotter matrix and then add the pull-type to be guest-pull there to save on the duplication?

File: .github/workflows/run-k8s-tests-on-zvsi.yaml (hunk 1 above)
Comment (BbolroC) [reply .github/workflows/run-k8s-tests-on-zvsi.yaml:98]: oh, good catch. I will update the PR. Thanks!

**Comment by BbolroC** [comment]
> The workflow for `run-k8s-tests-coco-nontee` and `run-k8s-tests` looks pretty much identical apart from the snapshotter and pull type, so I wonder if you could add the nydus snapshotter to the original snapshotter matrix and then add the pull-type to be guest-pull there to save on the duplication?

Yeah, I was thinking about that too. But it would be a matter of tee or non-tee. In the same sense, `run-k8s-tests-coco-nontee` and `run-k8s-tests` for `x86_64` are almost the same, but duplicated. I am sure if the deduplication is a way to go. What do you think @wainersm ?

File: .github/workflows/run-k8s-tests-on-zvsi.yaml [hunk 2]
```diff
@@ -31,18 +31,31 @@ jobs:
...
       DOCKER_REPO: ${{ inputs.repo }}
       DOCKER_TAG: ${{ inputs.tag }}
       PR_NUMBER: ${{ inputs.pr-number }}
+      GH_PR_NUMBER: ${{ inputs.pr-number }}
+      KATA_HOST_OS: "ubuntu"
       KATA_HYPERVISOR: ${{ matrix.vmm }}
       KUBERNETES: "k3s"
+      PULL_TYPE: ${{ matrix.pull-type }}
       SNAPSHOTTER: ${{ matrix.snapshotter }}
-      USING_NFD: "true"
```
Comment (wainersm) [inline .github/workflows/run-k8s-tests-on-zvsi.yaml:45]: I suspect this `USING_NFD` is bogus. IIRC @fidencio was preparing the workflows for the NFD (Node Feature Discovery) but it's unfinished. Please correct if I am wrong.

**Review by wainersm — APPROVED** [review]
Despite the USING_NFD comment that in reality is spread in many workflows; this looks good to me.

File: .github/workflows/run-k8s-tests-on-zvsi.yaml (hunk 2 above)
Comment (BbolroC) [reply .github/workflows/run-k8s-tests-on-zvsi.yaml:45]: Yeah, we could deal with it all at once later when it is finished. Thanks!

**Pull request merged**